cronenbroguelike/game.py

```diff
@@ -47,64 +47,68 @@ def execute(self):
         self.kill()
 
 
-def _start_game(config):
-    def _restart(unused_actor):
-
-        # Resets all global state (clears event queues, etc.).
-        _G.reset()
-
-        # Creates the player character and ensures game will restart upon death.
-        _G.player = actor.create_actor(
-            health=10,
-            psyche=10,
-            strength=10,
-            stamina=10,
-            name="player",
-        )
-        _G.player.log_stats = True
-
-        # Resets just_died flag.
-        _G.add_event(_ResetDiedFlag(), "pre")
+def _restart(config):
+    # Resets all global state (clears event queues, etc.).
+    _G.reset()
+
+    # Creates the player character and ensures game will restart upon death.
+    _G.player = actor.create_actor(
+        health=10,
+        psyche=10,
+        strength=10,
+        stamina=10,
+        name="player",
+    )
+    _G.player.log_stats = True
 
-        # Creates a small dungeon.
-        level = floor.Floor.generate("cathedral", config["num_rooms"])
+    # Resets just_died flag.
+    _G.add_event(_ResetDiedFlag(), "pre")
 
-        # Places a monster in a random room.
-        level.random_room().add_character(npcs.fish_man())
+    # Creates a small dungeon.
+    level = floor.Floor.generate("cathedral", config["num_rooms"])
 
-        # Places an NPC in a random room.
-        level.random_room().add_character(npcs.mad_librarian())
+    # Places a monster in a random room.
+    level.random_room().add_character(npcs.fish_man())
 
-        # Places a cool NPC in a random room.
-        level.random_room().add_character(npcs.smokes_man())
+    # Places an NPC in a random room.
+    level.random_room().add_character(npcs.mad_librarian())
 
-        # Places the player.
-        level.random_room().add_character(_G.player)
+    # Places a cool NPC in a random room.
+    level.random_room().add_character(npcs.smokes_man())
 
-        # Starts it up.
-        _get_random_start()
-        adventurelib.set_context("start_game")
-        adventurelib.set_context(None)
-        with _poll_events(poll_before=True, poll_after=True):
-            commands.enter_room(_G.player.current_room)
+    # Places the player.
+    level.random_room().add_character(_G.player)
 
-    _restart(None)
+    # Starts it up.
+    _get_random_start()
+    adventurelib.set_context("start_game")
+    adventurelib.set_context(None)
+    with _poll_events(poll_before=True, poll_after=True):
+        commands.enter_room(_G.player.current_room)
 
 
-def main():
-    game_config = _load_config()
-    if game_config.get("extra_commands") or game_config.get("random_run"):
+def run_game(config):
```

Comment:
Nitpick: `_run_game`

Reply:
I wanted to ask about this - is it mostly a convention for saying "please treat this as private"? Does anything check it? If I'm using `_run_game` elsewhere (like in the test) is it still good practices to prepend the `_` to signify that really you shouldn't?

Reply:
`pylint` (and maybe `black`?) check this stuff. It's a pretty strong convention in Python to do this, and "please treat this as private" is a good way to describe it, since the Python language itself of course has no concept of access control. But yeah, it's a good practice.